kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/CreateOrReplaceResourceTest.java

```diff
@@ -70,10 +75,21 @@ public void testResourceCreate() throws Exception {
   }
 
   @Test
-  public void testCreate() throws Exception {
-    server.expect().get().withPath("/api/v1/namespaces/test/pods/pod123").andReturn(404, new StatusBuilder().build()).always();
+  @DisplayName("Should throw Exception on failed create")
+  void testResourceCreateFailure() {
+    // Given
+    server.expect().post().withPath("/api/v1/namespaces/test/pods").andReturn(HttpURLConnection.HTTP_BAD_REQUEST, new PodBuilder()
+      .withNewMetadata().withResourceVersion("12345").endMetadata().build()).once();
+    KubernetesClient client = server.getClient();
 
-    server.expect().post().withPath("/api/v1/namespaces/test/pods").andReturn(201, new PodBuilder()
+    // When
+    assertThrows(KubernetesClientException.class, () -> client.resource(new PodBuilder().withNewMetadata().withName("pod123").endMetadata().build()).createOrReplace());
```

Comment:
You are getting sonar issues here. In order to make sure exception is thrown in the right place:
```suggestion
    Pod pod = client.resource(new PodBuilder().withNewMetadata().withName("pod123").endMetadata().build());
    // When
    assertThrows(KubernetesClientException.class, () -> pod.createOrReplace());
```

Same for the rest of warnings.